src/opensearch.ts

```diff
@@ -173,6 +173,80 @@ class Opensearch {
     }
   }
 
+  async getLatestApprovedReportsForWikidataId(wikidataId: string) {
+    try {
+      if (!this.client) throw new Error('Opensearch not connected')
+      console.log('fetching company report for wikidataId:', wikidataId)
+      const result = (await this.client.search({
+        index: this.indexName,
+        body: {
+          query: {
+            bool: {
+              must: [
+                {
+                  terms: {
+                    wikidataId: [wikidataId],
+                    state: ['approved'],
+                  },
+                },
+              ],
+            },
+          },
+          sort: [
+            {
+              timestamp: {
+                order: 'desc',
+              },
+            },
+          ],
+          size: 1000,
+          aggs: {
+            latest_reports: {
+              terms: {
+                field: 'report.wikidataId.keyword',
+                size: 1000,
+                order: {
+                  latest_timestamp: 'desc',
+                },
+              },
+              aggs: {
+                latest_timestamp: {
+                  max: {
+                    field: 'timestamp',
+                  },
+                },
+                latest_report: {
```

Comment:
Should this be `latest_report` or `latest_reports` like on line 204?

Reply:
Den ena summerar en lista och den andra tar ut den sista så det är korrekt